core/io/file_access.cpp

```diff
@@ -159,19 +159,24 @@ Error FileAccess::reopen(const String &p_path, int p_mode_flags) {
 Ref<FileAccess> FileAccess::open(const String &p_path, int p_mode_flags, Error *r_error) {
 	//try packed data first
 
+	Error err = OK;
 	Ref<FileAccess> ret;
 	if (!(p_mode_flags & WRITE) && !(p_mode_flags & SKIP_PACK) && PackedData::get_singleton() && !PackedData::get_singleton()->is_disabled()) {
 		ret = PackedData::get_singleton()->try_open_path(p_path);
 		if (ret.is_valid()) {
+			err = ret->get_error();
 			if (r_error) {
-				*r_error = OK;
+				*r_error = err;
+			}
+			if (err != OK) {
+				ret.unref();
 			}
 			return ret;
```

Comment:
```suggestion
			if (err == OK) {
				return ret;
			}
```
It should behave the same as if `nullptr` is returned, and try to fall back to raw file outside pack.

Reply:
I don't believe this is correct behavior. If `PackedData::get_singleton()->try_open_path` returns a FileAccess object, that means that the path is in PackedData and thus the canonical file location is in the pack. If trying to open that file in the pack returned an error, it means that there's something catastrophically wrong with our environment (wrong encryption key, the pck got deleted in the middle of running, etc.) and we should fail here.

Reply:
It should at least be consistent, so maybe:
- check if pack has file using `PackedData::get_singleton()->has_path`
  - if found, try loading and fail if `try_open_path` return `nullptr` or returned file has error set.
  - if not found, try using filesystem.

Reply:
Or `get_error` check can be moved to the `try_open_path` and it can return `nullptr` if error is set (to avoid double file map lookup).